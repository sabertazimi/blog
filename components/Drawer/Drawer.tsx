import type { DrawerProps } from 'antd';
import { Drawer as AntDrawer } from 'antd';

interface Props extends DrawerProps {}

const Drawer = ({ title, visible, onClose, children }: Props): JSX.Element => (
  <AntDrawer
    className="font-extrabold text-dark"
    placement="right"
    closable={false}
    title={title}
    visible={visible}
    onClose={onClose}
  >
    {children}
  </AntDrawer>
);

export default Drawer;
