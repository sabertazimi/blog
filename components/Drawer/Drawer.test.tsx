import { render } from '@testing-library/react';
import Drawer from './Drawer';

describe('Drawer', () => {
  test('should render correctly (snapshot)', () => {
    const { container } = render(
      <Drawer title="Drawer" visible={true} onClose={jest.fn()}>
        Drawer
      </Drawer>
    );

    expect(container).toMatchSnapshot();
  });
});
