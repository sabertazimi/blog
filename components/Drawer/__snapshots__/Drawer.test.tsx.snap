// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should render correctly (snapshot) 1`] = `
<div>
  <div>
    <div
      id="content"
    >
      <div
        class="ant-drawer ant-drawer-right ant-drawer-open font-extrabold text-dark"
        tabindex="-1"
      >
        <div
          class="ant-drawer-mask"
        />
        <div
          class="ant-drawer-content-wrapper"
          style="width: 378px;"
        >
          <div
            class="ant-drawer-content"
          >
            <div
              class="ant-drawer-wrapper-body"
            >
              <div
                class="ant-drawer-header"
              >
                <div
                  class="ant-drawer-header-title"
                >
                  <div
                    class="ant-drawer-title"
                  >
                    Drawer
                  </div>
                </div>
              </div>
              <div
                class="ant-drawer-body"
              >
                Drawer
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-target-tag-name="DIV"
      id="target"
    />
  </div>
</div>
`;
